### PR TITLE
[Components with more than 1 service] LPS-184060 Refactor Editor components from frontend-editor module

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://issues.liferay.com/browse/LPS-184060

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:frontend

**Comment by catarineaguiar** [comment]
@MrJohn1911 tests weren't tested due to:

```
     [exec] > Task :apps:frontend-editor:frontend-editor-alloyeditor-web:buildAlloyEditor
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/frontend-editor/frontend-editor-alloyeditor-web/src/main/java/com/liferay/frontend/editor/alloyeditor/web/internal/AlloyEditor.java:46: error: method does not override or implement a method from a supertype
     [exec] 	@Override
     [exec] 	^
     [exec] Note: /opt/dev/projects/github/liferay-portal/modules/apps/frontend-editor/frontend-editor-alloyeditor-web/src/main/java/com/liferay/frontend/editor/alloyeditor/web/internal/AlloyEditor.java uses or overrides a deprecated API.
     [exec] Note: Recompile with -Xlint:deprecation for details.
     [exec] Note: /opt/dev/projects/github/liferay-portal/modules/apps/frontend-editor/frontend-editor-alloyeditor-web/src/main/java/com/liferay/frontend/editor/alloyeditor/web/internal/editor/configuration/AlloyEditorCreoleConfigContributor.java uses unchecked or unsafe operations.
     [exec] Note: Recompile with -Xlint:unchecked for details.
     [exec] 1 error
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-37/test-portal-acceptance-pullrequest(master)/11498/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:frontend

**Comment by catarineaguiar** [comment]
ci:test:stable

**Comment by MrJohn1911** [comment]
ci:stop

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:frontend

**Comment by catarineaguiar** [comment]
ci:test:stable

**Comment by catarineaguiar** [comment]
@MrJohn1911 log failures when adding a basic web content:

```
2023-05-17 22:03:19.257 ERROR [http-nio-8080-exec-3][AlloyEditor:56] bundle com.liferay.frontend.editor.alloyeditor.web:5.0.38 (860)[com.liferay.frontend.editor.alloyeditor.web.internal.AlloyEditor(9828)] : The activate method has thrown an exception
java.lang.ClassCastException: [Ljava.lang.String; cannot be cast to java.lang.String
	at com.liferay.frontend.editor.alloyeditor.web.internal.AlloyEditor.activate(AlloyEditor.java:58) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_221]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_221]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_221]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_221]
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invokeMethod(BaseMethod.java:228) ~[?:?]
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.access$500(BaseMethod.java:41) ~[?:?]
	at org.apache.felix.scr.impl.inject.methods.BaseMethod$Resolved.invoke(BaseMethod.java:664) ~[?:?]

```
full log [here](https://testray.liferay.com/reports/production/logs/2023-05/test-1-31/test-portal-acceptance-pullrequest(master)/8057/functional-smoke-tomcat90-hypersonic20-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
Last commit was just a Baseline

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
ci:test:frontend

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/frontend-editor/frontend-editor-taglib/src/main/java/com/liferay/frontend/editor/taglib/internal/EditorRendererUtil.java [hunk 1]
```diff
@@ -48,12 +50,15 @@ public static void start(final BundleContext bundleContext) {
...
 					ServiceReference<EditorRenderer> serviceReference) {
 
-					String name = (String)serviceReference.getProperty("name");
-
 					EditorRenderer editorRenderer = bundleContext.getService(
 						serviceReference);
 
-					_editorRendererMap.put(name, editorRenderer);
+					List<String> propertyNames = StringUtil.asList(
+						serviceReference.getProperty("name"));
```
Comment (rafaprax) [inline modules/apps/frontend-editor/frontend-editor-taglib/src/main/java/com/liferay/frontend/editor/taglib/internal/EditorRendererUtil.java:57]: @MrJohn1911, please rename to `names`. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
In this last push, I just renamed a variable [here](https://github.com/liferay-core-infra/liferay-portal/pull/4542/commits/bc54ef6d8774ba648b6f9ef0a8dadf7a0446aaef) and solve a conflit with packageinfo.

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

File: modules/apps/frontend-editor/frontend-editor-alloyeditor-web/src/main/java/com/liferay/frontend/editor/alloyeditor/web/internal/AlloyEditorEditor.java [hunk 2]
```diff
@@ -28,47 +25,12 @@
...
-	public String getResourceType() {
-		return PortalWebResourceConstants.RESOURCE_TYPE_EDITOR_ALLOYEDITOR;
-	}
+@Component(property = "name=alloyeditor", service = Editor.class)
+public class AlloyEditorEditor extends BaseAlloyEditor {
 
 	@Activate
 	protected void activate(Map<String, Object> properties) {
-		_name = (String)properties.get("name");
+		setName((String)properties.get("name"));
```
Comment (rafaprax) [inline modules/apps/frontend-editor/frontend-editor-alloyeditor-web/src/main/java/com/liferay/frontend/editor/alloyeditor/web/internal/AlloyEditorEditor.java:33]: Hey @MrJohn1911, please create a constant called `EDITOR_NAME` with value `alloyeditor` and implement the `getName` and set the Component property as well. Also, get rid of the `setName` method. 

Please apply for the other cases as well. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:frontend

**Comment by Kvale1733** [comment]
@MrJohn1911 No regressions found.

**Comment by MrJohn1911** [comment]
In this last push, I just changed the property of the *EditorRenderer.java components from a hardcoded string to a constant.

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
ci:test:frontend

**Comment by zhangyan0701** [comment]
No regressions found.

**Review by rafaprax — APPROVED** [review]

**Pull request closed without merging**